Remove migration sticker

Imran92 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/92032

## Proposed Changes

Removes the migration sticker from the site when the user completes checkout or goes out of the flow. We've covered the usual flows-
- Unmounting/Navigating back from the "Upgrade" page
- Completing the checkout
- Clicking the button for importing the content only

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To remove the migration discount sticker from the site

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Enable the feature flag `migration-flow/introductory-offer`. You can add it to your `.env` file like this `ACTIVE_FEATURE_FLAGS="migration-flow/new-migration-instructions-step,migration-flow/introductory-offer"` 
- Start monitoring your browser requests.
- Start a migration and navigate until the step `site-migration-upgrade-plan`.
- Check that when that step loads, it requests the endpoint _.../{SITE_ID}/migration-flow_
- Now click the Back button on top left
- Make sure you see another request sent to the same URL as before, but it's a DELETE request
- Go back to the Migrate/Import page and click on the button for importing content only
- Make sure you see the Delete request being sent
- Now go to the upgrade page, select Upgrade and Migrate, and complete the checkout
- Make sure you see this call being made on the next page

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

renatho committed:
Add migration sticker wrapper component

renatho committed:
Add migration sticker wrapper component

renatho committed:
Add mutation to add sticker

renatho committed:
Add mutation to add sticker

renatho committed:
Fix request

renatho committed:
Fix request

renatho committed:
Add loading and error states

renatho committed:
Add loading and error states

Imran92 committed:
Add sticker deleting mutation

renatho committed:
Add loading state

renatho committed:
Unify upgrade plan loaders

So we don't have multiple loaders, causing screen flickering.

renatho committed:
Add introductory offer feature flag

renatho committed:
Add introductory offer feature flag

renatho committed:
Don't add sticker if feature flag is not enabled

merkushin committed:
Output upgrade plan details depending on the intro offer feature flag

Imran92 committed:
Merge branch 'add/migration-sticker' into remove/migration-sticker

renatho committed:
Extract upgrade plan loader component

renatho committed:
Remove not used files anymore

renatho committed:
Split parts in smaller components

renatho committed:
Transform wrapper in a HOC

renatho committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into add/migration-sticker

renatho committed:
Update HOC to be responsible only by migration sticker

renatho committed:
Improve check, so it accepts price 0

renatho committed:
Fix existing tests

merkushin committed:
Convert functions into components

merkushin committed:
Display introductory offer price

Imran92 committed:
Merge branch 'add/migration-sticker' into remove/migration-sticker

Imran92 committed:
Remove sticker if import selected

Imran92 committed:
Clear migration sticker on component unmount

renatho committed:
Add tests

Imran92 committed:
Delete sticker on after migration is complete

Imran92 committed:
Merge branch 'add/migration-sticker' into remove/migration-sticker

Imran92 committed:
Test delete sticker delete endpoint is called on unmount

Imran92 committed:
Test that clicking import calls delete migration sticker func

Imran92 committed:
Test that delete sticker function is called on instruction page load

renatho committed:
Fix tests

Imran92 committed:
Merge branch 'add/migration-sticker' into remove/migration-sticker

Imran92 committed:
Remove stuck changes

merkushin committed:
Get all needed data from different sources

merkushin committed:
Merge branch 'trunk' into add/migration-flow-display-intro-offer

merkushin committed:
Show proper billing time frame

Imran92 committed:
Remove sticker in new instruction page

Imran92 committed:
Add test for new migration instruction page

Imran92 committed:
Add feature flag check

merkushin committed:
Improve pricing offer

merkushin committed:
Fix badge styling

renatho commented on client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx:
```diff
@@ -20,7 +24,13 @@ const withMigrationSticker =
+
+			return () => {
+				if ( 0 !== siteId ) {
+					deleteMigrationSticker( siteId );
```
Just a double-check with @merkushin, we won't need the sticker in the checkout step for anything, right?

merkushin replied:
We need it until the end of the checkout. We can remove it when the plan was paid successfully.

Imran92 replied:
The "Upgrade" step does not get unmounted when `Checkout` opens as it opens in a modal. So the sticker won't get unset when we open the checkout.

renatho replied:
TIL! I thought it was a different step! Thank you for the answer! =)

renatho reviewed (approved):
Looks good and works well!

I'd just wait for the source branch to be merged before merging this, so the review doesn't get complicated with 2 PRs in one. =)

Also, I just added a comment for a double-check with Dmitry, just in case he explored something about the checkout that we don't know. 😉

merkushin committed:
Apply suggestions from code review

Co-authored-by: daledupreez <dale@automattic.com>

merkushin committed:
Use fallback translations for non-English users

- For new strings use existing fallback messages.
- Show the standard offer when the feature is enabled, but intro not
  available.
- Add tests.

merkushin committed:
Merge branch 'trunk' into add/migration-flow-display-intro-offer

merkushin committed:
Apply suggestions from code review

Co-authored-by: daledupreez <dale@automattic.com>

merkushin committed:
Revert removal of the flag

merkushin committed:
Extract a lightweigt price component

merkushin committed:
Update client/blocks/importer/wordpress/upgrade-plan/style.scss

Co-authored-by: daledupreez <dale@automattic.com>

merkushin committed:
Import type for TranslateResult and PropsWithChildren

Co-authored-by: daledupreez <dale@automattic.com>

merkushin committed:
Adjust styling

merkushin committed:
Merge branch 'trunk' into add/migration-flow-display-intro-offer

renatho committed:
Fix skeleton styles

merkushin committed:
Update tests for upgrade plan components

merkushin committed:
Fix test for upgrade plan without hoc

merkushin commented:
Hi @Imran92! There are a lot of conflicts it the PR. Could you solve them?

But I need to ask you to rebase your branch on `add/migration-flow-display-intro-offer` before that. In that branch, we already solved some conflicts and moved the wrapping from the UpgradePlanDetails component to UpgradePlan component (that's because fetching data moved to it). Maybe it is worth waiting for merging this [PR](https://github.com/Automattic/wp-calypso/pull/92201) first. 

Honestly, I'm not sure where it will be easier to solve the conflicts, but that PR is in priority for the project.

Commit:
Merge branch 'add/migration-flow-display-intro-offer' into remove/migration-sticker

Imran92 committed:
Remove unused style

Imran92 committed:
Remove loader component

Imran92 commented:
Thanks @merkushin, I've fixed all the conflicts. There were way too many conflicts, many of those were in files this PR doesn't event touch :p I wonder what happened that caused these many issues!

Imran92 committed:
Remove comment

renatho reviewed (approved):
Approving again! Thank you for the answer to my question!

merkushin commented:
@Imran92 I merged the PR for displaying the intro offer, that resulted in a new bunch of conflicts here.

> There were way too many conflicts, many of those were in files this PR doesn't event touch :p I wonder what happened that caused these many issues!

I think that happens because we always squash commits while merging. I just got a bunch of conflicts in another PR that didn't touch any JS, but the "original" branch also had some changes after branching out.

Imran92 committed:
Merge branch 'trunk' into remove/migration-sticker

Imran92 committed:
Fix import order

Imran92 committed:
Remove unnecessary change

Imran92 commented:
Fixed the conflicts again, as it was already approved, I'll go ahead and merge this to prevent more conflict issues

Imran92 committed:
Fix merge issue
